### ISSUE TITLE
We need documentation for how to use layouts.

### ISSUE THREAD

**Issue by codeviking** [allenai/varnish#112]
We're in the process of refactoring layouts, see allenai/varnish-legacy#158.  

That said once complete, we should write end user documentation for their usage.

**Comment by codeviking** [comment]
Ran into this again today, and nearly cut another issue (I'm doing Leaderboard work that requires mucking with the layout stuff).

We should think about simplifying the API too.  Here's what I had to do to make the layout left-anchored:

```jsx
<Layout.DefaultAppLayoutProvider appLayoutVariant="app">
  <Layout>
      ...
  </Layout>
</Layout.DefaultAppLayoutProvider />
```

What if we just made `<Layout />` include a `AppLayoutContextProvider`.  I'd also propose using `variant` instead of `appLayoutVariant` and making the options `left-anchored` and `horz-centered` (or something like that).  These options seem more explicit and easier for a user without context to understand. 

With these changes one could just do:

```
<Layout variant="left-anchored">
   ...
</Layout>
```

Lastly, I'd propose renaming `AppLayoutProvider` to just `LayoutConfig`.  Ideally this is an implementation detail that no one has to deal with -- but it seems like a better name that more concretely captures the intent.  It'd also make it easier to extend (by adding other configuration parameters) in the future.
